src/components/dapps/ModalCreateDapps.vue

```diff
@@ -284,32 +310,310 @@
 
+    const openOption = ref(false);
+
+    const unit_d = 3;
+    const decimal = 12;
+
+    const formData = reactive<FormData>({
+      endowment: bnToBn(plasmUtils.reduceDenomToBalance(27, unit_d, decimal)),
+      weight: new BN(200000000000),
+      projectName: '',
+      projectDesc: '',
+      webpage: '',
+    });
+
+    const selAccount = ref(0);
+    const toAccount = ref(props.allAccounts[0] as string);
+    const toAddress = ref(props.allAccounts[0] as string);
+    const toAccountName = ref(props.allAccountNames[0]);
+
+    watch(
+      selAccount,
+      () => {
+        toAccount.value = props.allAccounts[selAccount.value] as string;
+        toAccountName.value = props.allAccountNames[selAccount.value];
+        toAddress.value = props.allAccounts[selAccount.value] as string;
+
+        openOption.value = false;
+      },
+      { immediate: true }
+    );
+
+    const shortenToAddress = computed(() => {
+      const address = toAccount.value as string;
+      return `${address.slice(0, 6)}${'.'.repeat(6)}${address.slice(-6)}`;
+    });
+
+    //abi
+    const store = useStore();
+
+    const { api } = useApi();
+    const apiPromise: ApiPromise = api?.value as ApiPromise;
+
+    const { abi, onChangeAbi, onRemoveAbi } = useAbi();
+
+    const {
+      fileRef: wasmFromFile,
+      setFile: setWasmFile,
+      validate: isWasmFromFileValid,
+    } = useFile({
+      onChange: onChangeAbi,
+      onRemove: onRemoveAbi,
+      validate: (file) =>
+        file?.data.subarray(0, 4).toString() === '0,97,115,109',
+    });
+
+    const extensionFile = ['.contract', '.json'];
+
+    const onDropFile = (fileState: FileState) => {
+      setWasmFile(fileState);
+    };
+
+    const { wasm, isWasmValid } = useWasm(
+      wasmFromFile.value,
+      isWasmFromFileValid
+    );
+
+    const { onSend } = useSendTx();
+
+    const upload = async () => {
+      if (
+        formData.projectName === '' ||
+        toAddress.value === '' ||
+        wasm.value === null ||
+        !isWasmValid
+      ) {
+        store.dispatch(ActionTypes.SHOW_ALERT_MSG, {
+          msg: `Please check fields again`,
+          alertType: 'error',
+        });
+        return;
+      }
+
+      const abiData = abi.value as Abi | AnyJson;
+
+      console.log('s', abiData);
+      console.log('w', wasm.value);
+
+      let uploadTx: SubmittableExtrinsic<'promise'> | null = null;
+
+      try {
+        const code = new CodePromise(apiPromise, abiData, wasm.value);
+
+        //should be changable
+        // const unit_d = 3;
+        // const decimal = 12;
+        // const endowment = bnToBn(
+        //   plasmUtils.reduceDenomToBalance(27, unit_d, decimal)
+        // );
+        // const weight = new BN(200000000000);
+        ///
+        console.log('endowment', formData.endowment);
+        console.log('weight', formData.weight);
+
+        console.log('code', code);
+        console.log('method', code.tx);
+
+        const constructorIndex = 0;
```

Comment:
The constructor index doesn't seem to be user-definable like in the apps portal?
https://github.com/polkadot-js/apps/blob/0ece312038b78d0191a1747be78afb9342a64e17/packages/page-contracts/src/Codes/Upload.tsx#L189
If you are planning on fixing this for the next PR, please let me know.